### PR TITLE
[#54] Use pureconfig

### PR DESCRIPTION
**Pull request by Kabowyad**

### CODE REVIEW

**Comment by Som3** [comment]
Config is right, but where applying?

**Comment by Kabowyad** [comment]
@Som3 I added config, but i dunno where to apply, because the only place where i found loading conf is DefaultDB class 14 line and i couldn't apply it for.

File: core/src/main/scala/codesearch/core/config/Config.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package codesearch.core.config
+
+import codesearch.core.config.Config.DatabaseConfig
+import pureconfig.error.ConfigReaderFailures
+import pureconfig.loadConfig
+
+final case class Config(dataSourceClass: String, numThreads:Int, properties: DatabaseConfig)
+
+object Config {
+  private[config] final case class DatabaseConfig(port: Int, name: String, user: String, password: String)
```
Comment (Som3) [inline core/src/main/scala/codesearch/core/config/Config.scala]: `port` -> `portNumber`

File: core/src/main/resources/application.conf [hunk 2]
```diff
@@ -3,8 +3,11 @@ mydb = {
   properties = {
     portNumber = 5432
```
Comment (Som3) [inline core/src/main/resources/application.conf]: add placeholder variant

**Comment by Som3** [comment]
@Kabowyad  `> DefaultDB class 14 line and I couldn't apply it for.` why cant? If u cant - need refactoring code
otherwise this issue not useful

File: core/src/test/scala/ConfigSpec.scala [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import codesearch.core.config.DatabaseConfig
+import org.scalatest.{FlatSpec, Matchers}
+
+class ConfigSpec extends FlatSpec with Matchers {
+  val config = DatabaseConfig.load()
+
+  "Config" should "be not null" in {
+    config.right.get should not be null
```
Comment (ppressives) [inline core/src/test/scala/ConfigSpec.scala]: Right projection of either does not return null, it throws error if empty. Try something like 
```
config shouldBe Right(
  Config(
    DatabaseConfig(
    ....
    )
  )
)
```

File: core/src/test/resources/application.conf [hunk 4]
```diff
@@ -0,0 +1,13 @@
+mydb {
+  dataSourceClass = "org.postgresql.ds.PGSimpleDataSource"
+  properties {
+    portNumber = 5432
+    portNumber  = ${?PORT_NUMBER}
```
Comment (ppressives) [inline core/src/test/resources/application.conf]: Why not `DATABASE_PORT_NUMBER`?

**Pull request merged**